src/docfx/build/toc/TableOfContentsMapBuilder.cs

```diff
@@ -41,14 +43,31 @@ public void Add(Document tocFile, IEnumerable<Document> referencedDocuments, IEn
         /// Build toc map including all tocs and reversed toc mapping(document -> toc)
         /// </summary>
         /// <returns>The toc map</returns>
-        public TableOfContentsMap Build()
+        public TableOfContentsMap Build(Context context)
         {
             var documentToTocs = new Dictionary<Document, HashSet<Document>>();
 
             // reverse the mapping between toc and documents
             // order by toc path
             var allTocs = new List<Document>();
             var experimentalTocs = new List<Document>();
+
+            // handle conflicts
+            var tocsGroupBySiteUrl = _tocToDocuments.Keys.GroupBy(k => k.SiteUrl);
```

Comment:
Extract this logic to a method? As we discussed, this might be shared logic from the build output